### PR TITLE
MPI/CCL tests for distributed DBSCAN

### PR DESCRIPTION
**Pull request by amgrigoriev**

### CODE REVIEW

File: cpp/oneapi/dal/algo/dbscan/test/spmd_backend_template.hpp [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+    using float_t = std::tuple_element_t<0, TestType>;
+
+    const te::dataframe data = GENERATE_DATAFRAME(
+        te::dataframe_builder{ "workloads/hepmass/dataset/hepmass_10t_test.csv" });
+    const table x =
+        data.get_table(this->get_policy(), this->get_homogen_table_id(), sycl::usm::alloc::device);
+
+    constexpr double epsilon = 5;
+    constexpr std::int64_t min_observations = 3;
+    constexpr float_t ref_dbi = 0.78373;
```
Comment (makart19) [inline cpp/oneapi/dal/algo/dbscan/test/spmd_backend_template.hpp:361]:
```suggestion
    constexpr float_t ref_dbi = float_t(0.78373);
```

File: cpp/oneapi/dal/algo/dbscan/test/spmd_backend_template.hpp [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+
+    const te::dataframe data =
+        te::dataframe_builder{ "workloads/mnist/dataset/mnist_test.csv" }.build();
+
+    const table x =
+        data.get_table(this->get_policy(), this->get_homogen_table_id(), sycl::usm::alloc::device);
+
+    constexpr double epsilon = 1.7e3;
+    constexpr std::int64_t min_observations = 3;
+    constexpr float_t ref_dbi = 1.584515;
```
Comment (makart19) [inline cpp/oneapi/dal/algo/dbscan/test/spmd_backend_template.hpp:342]:
```suggestion
    constexpr float_t ref_dbi = float_t(1.584515);
```

File: cpp/oneapi/dal/algo/dbscan/test/spmd_backend_fixture.hpp [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        const auto centroids = te::centers_of_mass(data, responses, cluster_count);
+
+        auto dbi = te::davies_bouldin_index(data, centroids, responses);
+        CAPTURE(dbi, ref_dbi);
+        REQUIRE(check_value_with_ref_tol(dbi, ref_dbi, dbi_ref_tol));
+    }
+
+    bool check_value_with_ref_tol(float_t val, float_t ref_val, float_t ref_tol) {
+        float_t max_abs = std::max(fabs(val), fabs(ref_val));
+        if (max_abs == 0.0)
```
Comment (makart19) [inline cpp/oneapi/dal/algo/dbscan/test/spmd_backend_fixture.hpp:112]:
```suggestion
        if (max_abs == float_t(0))
```

**Review by makart19 — APPROVED** [review]

File: cpp/oneapi/dal/algo/dbscan/test/spmd_backend_template.hpp (hunk 1 above)
Comment (KulikovNikita) [reply cpp/oneapi/dal/algo/dbscan/test/spmd_backend_template.hpp:361]: @makart19 - it will be implicitly converted in any case.

**Review by KulikovNikita — APPROVED** [review]

**Pull request merged**